realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/SchemaTests.kt

```diff
@@ -0,0 +1,195 @@
+    fun disallowDestructiveUpdateOfSyncedRealm() {
+        for (operation in DestructiveSchemaOperation.values()) {
+            // Init schema
+            Realm.getInstance(config).close()
+            val className = "StringOnly"
+            val newClassName = "Dogplace"
+            val fieldName = "chars"
+            val newFieldName = "newchars"
+
+            DynamicRealm.getInstance(config).use { realm ->
+                assertTrue(realm.schema.contains(className))
+                val objectSchema = realm.schema[className]!!
+                assertNotNull(objectSchema)
+                assertTrue(objectSchema.hasField(fieldName))
+
+                realm.beginTransaction()
+                errorCollector.assertFailsWith<UnsupportedOperationException> {
+                    when (operation) {
+                        DestructiveSchemaOperation.REMOVE_CLASS ->
+                            realm.schema.remove(className)
+                        DestructiveSchemaOperation.RENAME_CLASS ->
+                            realm.schema.rename(className, newClassName)
+                        DestructiveSchemaOperation.SET_CLASS_NAME ->
+                            objectSchema.setClassName(newClassName)
+                        DestructiveSchemaOperation.REMOVE_FIELD ->
+                            objectSchema.removeField(fieldName)
+                        DestructiveSchemaOperation.RENAME_FIELD ->
+                            objectSchema.renameField(fieldName, newFieldName)
+                        DestructiveSchemaOperation.REMOVE_INDEX ->
+                            objectSchema.removeIndex(fieldName)
+                        DestructiveSchemaOperation.REMOVE_PRIMARY_KEY ->
+                            objectSchema.removePrimaryKey()
+                        DestructiveSchemaOperation.ADD_PRIMARY_KEY ->
+                            objectSchema.addPrimaryKey(fieldName)
+                        DestructiveSchemaOperation.ADD_FIELD_PRIMARY_KEY -> {
+                            objectSchema.addField(newFieldName, String::class.java, FieldAttribute.PRIMARY_KEY)
+                        }
+                    }
+                }
+                // Verify that operation is actually not performed in the transaction
+                assertTrue(realm.schema.contains(className))
+                assertFalse(realm.schema.contains(newClassName))
+                assertTrue(objectSchema.hasField(fieldName))
+                realm.cancelTransaction()
+
+                // Verify that operation is actually not performed after cancelling
+                assertTrue(realm.schema.contains(className))
+                assertFalse(realm.schema.contains(newClassName))
+                assertTrue(objectSchema.hasField(fieldName))
+                assertFalse(objectSchema.hasField(newFieldName))
+                assertNotNull(objectSchema)
+            }
+        }
+    }
+
+}
+
+inline fun <reified T> ErrorCollector.assertFailsWith(block : () -> Unit){
```

Comment:
Is this something we should as a general extension method? Perhaps move to `KotlinTestUtils` in that case then

Reply:
Moved to `KotlinTestUtils`